### PR TITLE
💄(frontend) update document summary UI

### PR DESCRIPTION
**Pull request by PanchoutNathan**
## Purpose
Integration of the new summary system.

## Proposal

- [x] Enhanced the document summary UI for better visibility and interaction.
- [x] Refactored the DocHeader and DocEditor components to improve layout and responsiveness.
- [x] Updated tests for the DocTableContent to reflect changes in heading interactions and visibility checks.
- [x] Delete dead code

### CODE REVIEW

File: src/frontend/apps/impress/src/features/docs/doc-editor/components/BlockNoteEditor.tsx [hunk 1]
```diff
@@ -23,8 +23,7 @@ const cssEditor = (readonly: boolean) => `
     height:100%
   };
   & .bn-editor {
-    padding-right: 30px;
-    ${readonly && `padding-left: 30px;`}
+  
   };
```
Comment (AntoLC) [inline src/frontend/apps/impress/src/features/docs/doc-editor/components/BlockNoteEditor.tsx:27]: You don't seem to need `.bn-editor` anymore.

```suggestion
```

File: src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+import { HeadingBlock, useEditorStore } from '@/features/docs/doc-editor';
+import { MAIN_LAYOUT_ID } from '@/layouts/conf';
+
+import { Heading } from './Heading';
+
+type DocSummaryProps = {
+  headings: HeadingBlock[];
+};
+
+export const DocSummary = ({ headings }: DocSummaryProps) => {
```
Comment (AntoLC) [inline src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx:15]: I initially called the feature `doc-summary`, then it was advised to change it to "Table of Content.", let's keep it like that so.
https://github.com/numerique-gouv/impress/commit/5bd78b80687a92818497df7d443f7bac252a1067


```suggestion
export const TableContent = ({ headings }: TableContentProps) => {
```

File: src/frontend/apps/impress/src/features/docs/doc-editor/components/DocEditor.tsx [hunk 3]
```diff
@@ -12,10 +13,10 @@ import { Versions, useDocVersion } from '@/features/docs/doc-versioning/';
 import { useResponsiveStore } from '@/stores';
 
 import { DocVersionHeader } from '../../doc-header/components/DocVersionHeader';
+import { DocSummary } from '../../doc-table-content/components/DocSummary';
```
Comment (AntoLC) [inline src/frontend/apps/impress/src/features/docs/doc-editor/components/DocEditor.tsx:16]:
```suggestion
import { DocSummary } from '@/features/docs/doc-table-content/';
```

File: src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+      $height="40px"
+      $zIndex={1000}
+      $align="center"
+      $padding="xs"
+      $justify="center"
+      $css={css`
+        border: 1px solid #ccc;
+        border: 1px solid #ccc;
+        overflow-x: hidden;
+        overflow-y: hidden;
```
Comment (AntoLC) [inline src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx:99]:
```suggestion
        border: 1px solid #ccc;
        overflow: hidden;
```

File: src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+          width: 200px;
+          height: auto;
+          max-height: calc(100vh - 60px - 15vh);
+        }
+      `}
+    >
+      {!isHover && (
+        <Box $justify="center" $align="center">
+          <Icon iconName="list" $theme="primary" $variation="800" />
+        </Box>
```
Comment (AntoLC) [inline src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx:119]: The color is strange on chrome at least:
image

Comment (AntoLC) [inline src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx:119]: Maybe add a small margin ? It is very close from the scroll bar no?
image

File: src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        setTimeout(() => {
+          const element = document.getElementById(
+            `heading-${headingIdHighlight}`,
+          );
+
+          element?.scrollIntoView({
+            behavior: 'smooth',
+            inline: 'center',
+            block: 'center',
+          });
```
Comment (AntoLC) [inline src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx:81]: The hover is not on the element where you clicked:

[scrnli_5OjEqYWaR2Im4U.webm](https://github.com/user-attachments/assets/94b6e87d-049e-42d1-a766-13b21ef6745e)

File: src/frontend/apps/impress/src/features/docs/doc-table-content/components/Heading.tsx [hunk 7]
```diff
@@ -47,23 +50,23 @@ export const Heading = ({
...
       $css="text-align: left;"
     >
       <Text
-        $theme="primary"
-        $padding={{ vertical: 'xtiny', left: 'tiny' }}
-        $size={sizeMap[level]}
+        $width="100%"
+        $padding={{ vertical: 'xtiny', left: leftPaddingMap[level] }}
+        $variation={isActive ? '1000' : '700'}
+        $weight={isActive ? 'bold' : 'normal'}
```
Comment (AntoLC) [inline src/frontend/apps/impress/src/features/docs/doc-table-content/components/Heading.tsx:68]: Long titles are out of container.

image

```suggestion
        $weight={isActive ? 'bold' : 'normal'}
        $css="overflow-wrap: break-word;" 
```

Or ellipsis ?

**Review by AntoLC — COMMENTED** [review]
Unrelated but you have an error from the `InfiniteScroll`:
image

---

Read only out of container:
image

Could be in another PR though.

File: src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx (hunk 5 above)
Comment (PanchoutNathan) [reply src/frontend/apps/impress/src/features/docs/doc-table-content/components/DocSummary.tsx]: I can't reproduce with 3 different browsers, is it possible to have your config?

**Review by AntoLC — APPROVED** [review]

**Pull request merged**